pkg/kubelet/status/status_manager_test.go

```diff
@@ -133,10 +150,37 @@ func (m *manager) consumeUpdates() int {
 	}
 }
 
+func TestInvalidNewStatus(t *testing.T) {
+	syncer := newTestManager(&fake.Clientset{})
+	testPod := getTestPod()
+	podStatus := getPodStatusWithRandomMessage()
+	if !syncer.SetPodStatus(testPod, podStatus) {
+		t.Errorf("SetPodStatus should have succeeded")
+	}
+	newPodStatus := getPodStatusWithRandomMessage()
+	newPodStatus.InitContainerStatuses[0].State.Terminated = nil
+	if syncer.SetPodStatus(testPod, newPodStatus) {
+		t.Errorf("SetPodStatus should have failed due to illegal transition for InitContainerStatus")
+	}
+	verifyUpdates(t, syncer, 1)
```

Comment:
Nit: I think we want this `verifyUpdates` check to occur at the end? Having it at the end tests that, given the four state transformations we attempted (three of which were invalid), only the one valid one actually caused an update?

Reply:
I have moved the verifyUpdates call to the end of the test.